RoutesAndTimetables/RoutesAndTimetables/RoutesAndTimetables/View/MainPage.xaml.cs

```diff
@@ -0,0 +1,30 @@
+﻿using RoutesAndTimetables.Business.DataObjects;
+using RoutesAndTimetables.Business.ViewModels;
+using RoutesAndTimetables.Ui.Services;
+using System;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace RoutesAndTimetables.Ui.View
+{
+    public partial class MainPage : ContentPage
+    {
+        public MainPage()
+        {
+            InitializeComponent();
+            
+            BindingContext = new MainPageViewModel(new NavigationService());
+        }
+
+
+        private void RouteTapped(object sender, ItemTappedEventArgs e)
+        {
+            (BindingContext as MainPageViewModel).RouteTappedCommand.Execute(e.Item as Route);
```

Comment:
You could create a property for `(BindingContext as MainPageViewModel)`, like `private MainPageViewModel ViewModel { get { return (BindingContext as MainPageViewModel); } }` for clarity, this would avoid always casting the binding context whenever you need to call your view model